File: 백준/Bronze/13223. 소금 폭탄/소금 폭탄.java
import java.io.*;
import java.util.*;

public class Main {
    static FastReader scan = new FastReader();
    static PrintWriter out = new PrintWriter(System.out);

    private static int timeToSeconds(String time) {
        String[] parts = time.split(":");
        int hours = Integer.parseInt(parts[0]);
        int minutes = Integer.parseInt(parts[1]);
        int seconds = Integer.parseInt(parts[2]);
        return hours * 3600 + minutes * 60 + seconds;
    }

    private static String secondsToTime(int sec) {
        int hours = sec / 3600;
        int minutes = (sec % 3600) / 60;
        int seconds = sec % 60;
        return String.format("%02d:%02d:%02d", hours, minutes, seconds);
    }

    public static void main(String[] args) throws IOException {
        String currTime = scan.nextLine();
        String targetTime = scan.nextLine();

        // 초 단위로 변환
        int currTimeSec = timeToSeconds(currTime);
        int targetTimeSec = timeToSeconds(targetTime);

        if (currTimeSec < targetTimeSec) {
            out.println(secondsToTime(targetTimeSec - currTimeSec));
        } else {
            out.println(secondsToTime(targetTimeSec + 24 * 60 * 60 - currTimeSec));
        }

        scan.close();
        out.close();
    }

    static class FastReader {
        private final BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        private final StreamTokenizer tokenizer = new StreamTokenizer(reader);

        public String nextString() throws IOException {
            tokenizer.nextToken();
            return tokenizer.sval;
        }

        public int nextInt() throws IOException {
            tokenizer.nextToken();
            return (int) tokenizer.nval;
        }

        public String nextLine() throws IOException {
            return reader.readLine();
        }

        void close() throws IOException {
            reader.close();
        }
    }
}

File: 백준/Bronze/13223. 소금 폭탄/소금 폭탄.kt
import java.io.*

private val scan = FastReader()
private val out = PrintWriter(System.out)

private var currTime = ""
private var targetTime = ""

fun timeToSeconds(time: String): Int {
    val parts: List<Int> = time.split(":").map { it.toInt() }
    val hours = parts[0]
    val minutes = parts[1]
    val seconds = parts[2]
    return hours * 3600 + minutes * 60 + seconds
}

fun secondsToTime(sec: Int): String {
    val hours = sec / 3600
    val minutes = (sec % 3600) / 60
    val seconds = sec % 60
    return String.format("%02d:%02d:%02d", hours, minutes, seconds)
}

fun main() {
    currTime = scan.nextLine()
    targetTime = scan.nextLine()
    // 초 단위로 변환
    val currTimeSec = timeToSeconds(currTime)
    val targetTimeSec = timeToSeconds(targetTime)

    if (currTimeSec < targetTimeSec) {
        out.println(secondsToTime(targetTimeSec - currTimeSec))
    } else {
        out.println(secondsToTime(targetTimeSec + 24 * 60 * 60 - currTimeSec))
    }

    scan.close()
    out.close()
}

class FastReader() {
    private val reader = BufferedReader(InputStreamReader(System.`in`))
    private val tokenizer = StreamTokenizer(reader)

    fun nextString(): String {
        tokenizer.nextToken()
        return tokenizer.sval
    }

    fun nextInt(): Int {
        tokenizer.nextToken()
        return tokenizer.nval.toInt()
    }

    fun nextLine(): String {
        return reader.readLine()
    }

    fun close() {
        reader.close()
    }
}